internal/pkg/docker/orchestrator/orchestrator.go

```diff
@@ -434,16 +464,19 @@ func (o *Orchestrator) stopTask(ctx context.Context, task Task) error {
 }
 
 // waitForContainerToStart blocks until the container specified by id starts.
-func (o *Orchestrator) waitForContainerToStart(ctx context.Context, id string) error {
+func (o *Orchestrator) waitForContainerToStart(ctx context.Context, id string, isEssential bool) error {
 	for {
 		isRunning, err := o.docker.IsContainerRunning(ctx, id)
 		switch {
 		case err != nil:
+			var errContainerExited *dockerengine.ErrContainerExited
+			if errors.As(err, &errContainerExited) && !isEssential {
```

Comment:
do you know what ECS does when a non-essential container fails to start?

Reply:
In case if `Essential Container Exits` the deployment circuit breaker will be triggered. But in case if `Non Essential container exits` ECS will not trigger the circuit breaker and continue to deploy the service.

This is what i see in the Console for NonEssential containers.

<img width="1361" alt="Screenshot 2023-12-15 at 1 40 26 PM" src="https://github.com/aws/copilot-cli/assets/71282729/6e80cfba-84d3-455a-b505-695890575cbc">

Reply:
yeah I know ECS doesn't error out or anything if the non-essential container _has started and exited_. I am curious what happens if the container was never able to start. I think it is not super important for us though, just in case you knew on top of you mind (if you don't, don't bother testing it 😂 !)